Library/Homebrew/formula.rb

```diff
@@ -927,6 +927,40 @@ def unlock
   end
 
   # @private
+  def outdated_versions
+    all_versions = []
+    older_or_same_tap_versions = []
+
+    if oldname && !rack.exist? && (dir = HOMEBREW_CELLAR/oldname).directory? &&
+      !dir.subdirs.empty? && tap == Tab.for_keg(dir.subdirs.first).tap
+      raise Migrator::MigrationNeededError.new(f)
+    end
+
+    rack.subdirs.each do |keg_dir|
```

Comment:
If you want, this PR offers us a chance to do more deep refactoring. e.g.
* make  `rack.subdirs` to `Formula#keg_dirs`
* introduce `Formula#kegs`

So we can avoid more duplicated logics.